### PR TITLE
With many on-demand cores, performance in order processing is subpar

### PR DESCRIPTION
**Pull request by eskimor**
This is a first rudimentary fix. Actual implementation perf improvements are no their way.

This also includes a fix where we used max instead of min: Will become significant once we have >1 claimqueues.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5056241

File: polkadot/primitives/src/v6/mod.rs [hunk 1]
```diff
@@ -393,7 +393,7 @@ pub const MAX_POV_SIZE: u32 = 5 * 1024 * 1024;
 /// Default queue size we use for the on-demand order book.
 ///
 /// Can be adjusted in configuration.
-pub const ON_DEMAND_DEFAULT_QUEUE_MAX_SIZE: u32 = 10_000;
+pub const ON_DEMAND_DEFAULT_QUEUE_MAX_SIZE: u32 = 500;
```
Comment (sandreim) [inline polkadot/primitives/src/v6/mod.rs:396]: Is this sufficient to not build overweight blocks ?

File: polkadot/runtime/parachains/src/scheduler.rs [hunk 2]
```diff
@@ -584,7 +584,8 @@ impl<T: Config> Pallet<T> {
 			return
 		}
 		// If there exists a core, ensure we schedule at least one job onto it.
-		let n_lookahead = Self::claimqueue_lookahead().max(1);
+		// TODO (now): Add a test for this. This wax `max` and nobody noticed.
+		let n_lookahead = Self::claimqueue_lookahead().min(1);
```
Comment (bkchr) [inline polkadot/runtime/parachains/src/scheduler.rs:588]: This was max out of the reason to always have at least 1 lookahead.

Comment (bkchr) [reply polkadot/runtime/parachains/src/scheduler.rs:588]: Because `0` makes no sense.

File: polkadot/primitives/src/v6/mod.rs (hunk 1 above)
Comment (bkchr) [reply polkadot/primitives/src/v6/mod.rs:396]: This changes will not fix anything. This variable is only used for migrations or at genesis. If you want to lower this, you need to set this in the `HostConfiguration` on chain.

File: polkadot/runtime/parachains/src/scheduler.rs (hunk 2 above)
Comment (eskimor) [reply polkadot/runtime/parachains/src/scheduler.rs:588]: LOL. Just realizing I was after all holding it wrong. Thanks.

**Comment by eskimor** [comment]
@antonva closing this one, as nothing is really left of this insubstantial PR anyways. Can you make sure queue size is reduced effectively. The default is still a good idea to also change, but this will already be part of my bigger PR anyways.

**Pull request closed without merging**